docs/networking/stun.md

```diff
@@ -122,3 +122,53 @@ originate. Using these internal addresses is much more likely to result in a
 successful direct connection.
 
 ![Diagram of a workspace agent and client over VPN](../images/networking/stun3.png)
+
+## Hard NAT
+
+Some NATs are known to use a different port when forwarding requests to the STUN
+server and when forwarding probe packets to peers. In that case, the address a
+peer discovers over the STUN protocol will have the correct IP address, but the
+wrong port. Tailscale refers to this as "hard" NAT in
+[How NAT traversal works (tailscale.com)](https://tailscale.com/blog/how-nat-traversal-works).
+
+If both peers are behind a "hard" NAT, direct connections may take longer to
+establish or will not be established at all. If one peer is behind a "hard" NAT
+and the other is running a firewall (including Windows Defender Firewall), the
+firewall may block direct connections.
+
+In both cases, peers fallback to DERP connections if they cannot establish a
+direct connection.
+
+If your workspaces are behind a "hard" NAT, you can:
+
+1. Ensure clients are not also behind a "hard" NAT. You may have limited ability
+   to control this if end users connect from their homes.
+2. Ensure firewalls on client devices (e.g. Windows Defender Firewall) have an
+   inbound policy allowing all UDP ports either to the `coder` or `coder.exe`
+   CLI binary, or from the IP addresses of your workspace NATs.
```

Comment:
Maybe we should list where the binary is commonly found on each platform.

- Linux: `/usr/bin/coder`
- Mac: unsure where brew puts it
- Windows: `C:\Program Files\Coder\bin\coder.exe` or `"C:\Users\dean\AppData\Roaming\Code\User\globalStorage\coder.coder-remote\bin\coder-windows-$ARCH.exe"`